app/src/androidTest/java/org/mozilla/fenix/ui/robots/DownloadRobot.kt

```diff
@@ -133,16 +133,36 @@ fun downloadRobot(interact: DownloadRobot.() -> Unit): DownloadRobot.Transition
 }
 
 private fun assertDownloadPrompt(fileName: String) {
-    assertTrue(
-        "Download prompt button not visible",
-        mDevice.findObject(UiSelector().resourceId("$packageName:id/download_button"))
-            .waitForExists(waitingTime)
-    )
-    assertTrue(
-        "$fileName title doesn't match",
-        mDevice.findObject(UiSelector().text(fileName))
-            .waitForExists(waitingTime)
-    )
+    var currentTries = 0
+    while (currentTries++ < 3) {
+        try {
+            assertTrue(
+                "Download prompt button not visible",
+                mDevice.findObject(UiSelector().resourceId("$packageName:id/download_button"))
+                    .waitForExists(waitingTime)
+            )
+            assertTrue(
+                "$fileName title doesn't match",
+                mDevice.findObject(UiSelector().text(fileName))
+                    .waitForExists(waitingTime)
+            )
+        } catch (e: AssertionError) {
+            browserScreen {
+            }.clickDownloadLink(fileName) {
+            }
+
+            assertTrue(
+                "Download prompt button not visible",
+                mDevice.findObject(UiSelector().resourceId("$packageName:id/download_button"))
+                    .waitForExists(waitingTime)
+            )
+            assertTrue(
+                "$fileName title doesn't match",
+                mDevice.findObject(UiSelector().text(fileName))
+                    .waitForExists(waitingTime)
+            )
```

Comment:
These duplicate asserts are not needed if we're using retry loop. The `catch` logic you've written will re-attempt the `click` on the download link, and iterate again in the `while` loop. The next iteration will `Assert` again and hopefully `break` to exit.

Reply:
Yes, you are right.